__tests__/app.test.js

```diff
@@ -186,5 +186,88 @@ describe('GET /api/articles/:article_id/comments', () => {
    });
 
 
+   test('POST 201: Responds with created comment upon successful POST', () => {
+    let commentToAdd = {
+      username: 'rogersop',
+      body: 'Foo bar'
+    }
+    return request(app)
+    .post('/api/articles/1/comments')
+    .send(commentToAdd)
+    .expect(201)
+    .then(( result ) => { 
+      expect(result.body.comment).toMatchObject({
+        author: 'rogersop',
+        body: 'Foo bar',
+        article_id: 1
+      })
+      expect(result.body.comment.hasOwnProperty("votes")).toBe(true);
+      expect(result.body.comment.hasOwnProperty("created_at")).toBe(true);
+    })  
+   });
+
+
+   test('POST 400: Responds with error when passed a body with missing information', () => {
+    let commentToAdd = {
+      username: 'rogersop'
+    }
+    return request(app)
+    .post('/api/articles/1/comments')
+    .send(commentToAdd)
+    .expect(400)
+    .then(( {body} ) => { 
+      expect(body.msg).toBe("Missing required information")
+    })  
+   });
+
+
+   test('POST 400: Responds with error when article id which is valid but non-existent', () => {
+    let commentToAdd = {
+      username: 'rogersop',
+      body: 'Foo bar'
+    }
+    return request(app)
+    .post('/api/articles/99999/comments')
+    .send(commentToAdd)
+    .expect(404)
+    .then(( {body} ) => { 
+      expect(body.msg).toBe("Article ID not found")
+    })  
+   });
+
+
+   test('POST 400: Responds with error when given invalid article id', () => {
+    let commentToAdd = {
+      username: 'rogersop',
+      body: 'Foo bar'
+    }
+    return request(app)
+    .post('/api/articles/not-an-article-id/comments')
+    .send(commentToAdd)
+    .expect(400)
+    .then(({ body }) => {  
+    expect(body.msg).toBe("Invalid article ID")
+    })
+   });
+
+   test("POST: 400 responds with error message when given a non-existent username", () => {
```

Comment:
Please change this test to expect 404 as the username is not in the database currently, but it could be added anytime, just like other data.